snuba/migrations/snuba_migrations/generic_metrics/0001_sets_aggregate_table.py

```diff
@@ -0,0 +1,143 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import (
+    AggregateFunction,
+    Array,
+    Column,
+    DateTime,
+    Nested,
+    String,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.storages.tags_hash_map import (
+    hash_map_int_column_definition,
+    hash_map_int_key_str_value_column_definition,
+)
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    granularity = "2048"
+    local_table_name = "generic_metric_sets_local"
+    columns: Sequence[Column[Modifiers]] = [
+        Column("use_case_id", String(Modifiers(low_cardinality=True))),
+        Column("org_id", UInt(64)),
+        Column("project_id", UInt(64)),
+        Column("metric_id", UInt(64)),
+        Column("granularity", UInt(8)),
+        Column("timestamp", DateTime()),
+        Column("retention_days", UInt(16)),
+        Column("indexed_tags", Nested([("key", UInt(64)), ("value", UInt(64))])),
+        Column("raw_tags", Nested([("key", UInt(64)), ("value", String())])),
+        Column("value", AggregateFunction("uniqCombined64", [UInt(64)])),
+        Column("timeseries_id", UInt(64)),
```

Comment:
`timeseries_id` will be a hash generated by the consumer for sharding so we can isolate that logic from ClickHouse itself

Reply:
You should not need this in the aggregated merge tree. Only in the raw table.
Sharding happens at the raw table level. 
- you write on the raw table
- the distributed raw table partitions data across shards according to the sharding key
- each storage node writes data on the raw table locally
- the materialized view generates the aggregated data and stores them in the aggregatingMergeTree locally (from the local raw table). There is no sharding happening at this step.

(also I think we will be able to get away with 32 bit for the time being, since it is data retained for a short period of time it will not be a problem.)